cpu/stm32/include/periph/cpu_common.h

```diff
@@ -113,6 +113,17 @@ typedef enum {
 #endif
 } bus_t;
 
+/**
+ * @brief STM32 peripherial base type.
+ *
+ * This structure contains things that are common to all STM32 peripherals.
+ */
+typedef struct
+{
+    volatile uint32_t *en_reg;      /**< RCC enable reg */
+    uint32_t en_mask;               /**< RCC enable reg mask */
+} periph_base_t;
```

Comment:
While every/most periph need to be clocked via the rcc this is certainly not the base of that peripheral, but more like a "RCC configuration" of said periph. (`rcc_config_t` or `periph_clock_config_t`)

If the base of an periph would be pointed at somewhere, I would expect is to point to that periphs base adress eg: for timer 1 to TIM1_CR1 (add an offset to that base and you find the other registers)